src/axom/inlet/tests/inlet_object.cpp

```diff
@@ -70,49 +77,52 @@ TEST(inlet_object, simple_struct_by_value)
   EXPECT_FALSE(foo.baz);
 }
 
-TEST(inlet_object, simple_array_of_struct_by_value)
+TYPED_TEST(InletObjectTest, simple_array_of_struct_by_value)
 {
   std::string testString =
-    "foo = { [4] = { bar = true; baz = false}, "
-    "        [7] = { bar = false; baz = true} }";
+    "foo = { [0] = { bar = true; baz = false}, "
+    "        [1] = { bar = false; baz = true} }";
   DataStore ds;
-  auto inlet = createBasicInlet(&ds, testString);
+  Inlet inlet = createBasicInlet<TypeParam>(&ds, testString);
 
   auto& arr_table = inlet.addGenericArray("foo");
 
   arr_table.addBool("bar", "bar's description");
   arr_table.addBool("baz", "baz's description");
-  std::unordered_map<int, Foo> expected_foos = {{4, {true, false}},
-                                                {7, {false, true}}};
+  std::unordered_map<int, Foo> expected_foos = {{0, {true, false}},
+                                                {1, {false, true}}};
   auto foos = inlet["foo"].get<std::unordered_map<int, Foo>>();
   EXPECT_EQ(foos, expected_foos);
 }
 
-TEST(inlet_object, simple_array_of_struct_implicit_idx)
+TYPED_TEST(InletObjectTest, simple_array_of_struct_implicit_idx)
 {
   std::string testString =
     "foo = { { bar = true; baz = false}, "
     "        { bar = false; baz = true} }";
   DataStore ds;
-  auto inlet = createBasicInlet(&ds, testString);
+  Inlet inlet = createBasicInlet<TypeParam>(&ds, testString);
 
   auto& arr_table = inlet.addGenericArray("foo");
 
   arr_table.addBool("bar", "bar's description");
   arr_table.addBool("baz", "baz's description");
-  std::unordered_map<int, Foo> expected_foos = {{1, {true, false}},
-                                                {2, {false, true}}};
+  // Lua is 1-indexed
+  const int base_idx =
```

Comment:
Suggestion: Seems like it might be a good idea for the reader to define the `base_idx`